File: flask1/Controller/FriendController.py
from flask import current_app as app

from unittest import result
from flask import Blueprint
from flask import request

from flask import  session
from flask.views import MethodView

from dbExts import db

from Model.FriendModel import FriendModel


# 定义蓝图
bp = Blueprint('blueprint', __name__)

class FriendController(MethodView):
    def get_friend_list():
        data = request.get_json()
        start = data.get('start')
        num = data.get('num')
            
        # start = "0"
        # num = "3"

        if  not num:
            return { 'result': app.config['FAILED'], 'msg': "信息不全" }

        ret = FriendModel.get_friend_list(start, num)

        return ret 

    def add_friend():
        data = request.get_json()
        uid = data.get('uid')
        data1 = data.get('data')
        
        # start = "0"
        # num = "3"

        if  not uid:
            return { 'result': app.config['FAILED'], 'msg': "信息不全" }

        ret = FriendModel.add_friend(data1)

        return ret 

    def update_friend():
        data = request.get_json()
        uid = data.get('uid')
        data1 = data.get('data')
        
        # start = "0"
        # num = "3"

        if  not uid:
            return { 'result': app.config['FAILED'], 'msg': "信息不全" }

        ret = FriendModel.update_friend(data1)

        return ret 

    def delete_friend():
        data = request.get_json()
        uid = data.get('uid')
        data1 = data.get('data')
   
        if  not uid:
            return { 'result': app.config['FAILED'], 'msg': "信息不全" }

        ret = FriendModel.delete_friend(data1)

        return ret 

File: gin1/controller/FriendController.go
package controller


import (
	"github.com/gin-gonic/gin"
	"net/http"
	// "fmt"

	"gin1/model"

)

type FriendController struct{
}


func (uc *FriendController) Get_friend_list(c *gin.Context) {

	var jsonData map[string]interface{}

	if err := c.ShouldBindJSON(&jsonData); err != nil {
		c.JSON(400, gin.H{"error": err.Error()})
		return
	}
 
	start := int(jsonData["start"].(float64))
	num := int(jsonData["num"].(float64))

	// 调用模型
	friendModel := &model.FriendModel{}
	ret := friendModel.Get_friend_list(start, num)

	c.JSON(http.StatusOK, ret)
}


func (uc *FriendController) Add_friend(c *gin.Context) {

	var jsonData map[string]interface{}

	if err := c.ShouldBindJSON(&jsonData); err != nil {
		c.JSON(400, gin.H{"error": err.Error()})
		return
	}
 
	data := jsonData["data"]

	// 调用模型
	friendModel := &model.FriendModel{}
	ret := friendModel.Add_friend(data)

	c.JSON(http.StatusOK, ret)
}


func (uc *FriendController) Update_friend(c *gin.Context) {

	var jsonData map[string]interface{}

	if err := c.ShouldBindJSON(&jsonData); err != nil {
		c.JSON(400, gin.H{"error": err.Error()})
		return
	}
 
	data := jsonData["data"]

	// 调用模型
	friendModel := &model.FriendModel{}
	ret := friendModel.Update_friend(data)

	c.JSON(http.StatusOK, ret)
}

func (uc *FriendController) Delete_friend(c *gin.Context) {

	var jsonData map[string]interface{}

	if err := c.ShouldBindJSON(&jsonData); err != nil {
		c.JSON(400, gin.H{"error": err.Error()})
		return
	}
 
	data := jsonData["data"]

	// 调用模型
	friendModel := &model.FriendModel{}
	ret := friendModel.Delete_friend(data)

	c.JSON(http.StatusOK, ret)
}